server/command.go

```diff
@@ -95,10 +95,205 @@ func (p *Plugin) postCommandResponse(args *model.CommandArgs, text string) {
+	if len(parameters) == 0 {
+		p.postCommandResponse(args, "Please specify a repository.")
+		return &model.CommandResponse{}, nil
+	}
+
+	repo := parameters[0]
+
+	if err := p.Unsubscribe(args.ChannelId, repo); err != nil {
+		mlog.Error(err.Error())
+		p.postCommandResponse(args, "Encountered an error trying to unsubscribe. Please try again.")
+		return &model.CommandResponse{}, nil
+	}
+
+	p.postCommandResponse(args, fmt.Sprintf("Succesfully unsubscribed from %s.", repo))
+	return &model.CommandResponse{}, nil
+}
+func handleDisconnect(p *Plugin, _ *plugin.Context, args *model.CommandArgs, _ []string, _ *github.Client, _ context.Context, _ *GitHubUserInfo) (*model.CommandResponse, *model.AppError) {
+	p.disconnectGitHubAccount(args.UserId)
+	p.postCommandResponse(args, "Disconnected your GitHub account.")
+	return &model.CommandResponse{}, nil
+}
+func handleTodo(p *Plugin, _ *plugin.Context, args *model.CommandArgs, _ []string, githubClient *github.Client, ctx context.Context, userInfo *GitHubUserInfo) (*model.CommandResponse, *model.AppError) {
+	text, err := p.GetToDo(ctx, userInfo.GitHubUsername, githubClient)
+	if err != nil {
+		mlog.Error(err.Error())
+		p.postCommandResponse(args, "Encountered an error getting your to do items.")
+		return &model.CommandResponse{}, nil
+	}
+	p.postCommandResponse(args, text)
+	return &model.CommandResponse{}, nil
+}
+func handleMe(p *Plugin, _ *plugin.Context, args *model.CommandArgs, _ []string, githubClient *github.Client, ctx context.Context, _ *GitHubUserInfo) (*model.CommandResponse, *model.AppError) {
+	gitUser, _, err := githubClient.Users.Get(ctx, "")
+	if err != nil {
+		p.postCommandResponse(args, "Encountered an error getting your GitHub profile.")
+		return &model.CommandResponse{}, nil
+	}
+
+	text := fmt.Sprintf("You are connected to GitHub as:\n# [![image](%s =40x40)](%s) [%s](%s)", gitUser.GetAvatarURL(), gitUser.GetHTMLURL(), gitUser.GetLogin(), gitUser.GetHTMLURL())
+	p.postCommandResponse(args, text)
+	return &model.CommandResponse{}, nil
+}
+func handleHelp(p *Plugin, _ *plugin.Context, args *model.CommandArgs, _ []string, _ *github.Client, _ context.Context, _ *GitHubUserInfo) (*model.CommandResponse, *model.AppError) {
+	text := "###### Mattermost GitHub Plugin - Slash Command Help\n" + strings.Replace(COMMAND_HELP, "|", "`", -1)
+	p.postCommandResponse(args, text)
+	return &model.CommandResponse{}, nil
+}
+func handleEmpty(p *Plugin, _ *plugin.Context, args *model.CommandArgs, _ []string, _ *github.Client, _ context.Context, _ *GitHubUserInfo) (*model.CommandResponse, *model.AppError) {
+	text := "###### Mattermost GitHub Plugin - Slash Command Help\n" + strings.Replace(COMMAND_HELP, "|", "`", -1)
+	p.postCommandResponse(args, text)
+	return &model.CommandResponse{}, nil
+}
+func handleSettings(p *Plugin, _ *plugin.Context, args *model.CommandArgs, parameters []string, _ *github.Client, _ context.Context, userInfo *GitHubUserInfo) (*model.CommandResponse, *model.AppError) {
```

Comment:
Why is this a function and not a method of `*Plugin`?

Reply:
In that case, `commandHandlers` map should be inside `Plugin` and initialize with it. Something like that
```go
p.commandHandlers = map[string]CommandHandleFunc{
		"subscribe":   p.handleSubscribe,
		"unsubscribe": p.handleUnSubscribe,
		"disconnect": p.handleDisconnect,
		"todo":        p.handleTodo,
		"me":          p.handleMe,
		"help":       p.handleHelp,
		"":            p.handleEmpty,
		"settings":    p.handleSettings,
	}
```